### ISSUE TITLE
False Positive: Squiz.Arrays.ArrayDeclaration.ValueNoNewline

### ISSUE THREAD

**Issue by simPod** [squizlabs/PHP_CodeSniffer#3952]
## Describe the bug

Squiz.Arrays.ArrayDeclaration.ValueNoNewline gives false positives

### Code sample
```php
$x = [
    static fn (  // false positive here
        A $a,
    ) => B::create(
        $a,
    ),
    false,
];

$y = [
    ...[ // false positive here
        new A(),
    ],
];
```

### To reproduce
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```
PHPCS output here
```

## Expected behavior
No rrror

## Versions (please complete the following information)

| | |
|-|-|
| Operating System | [MacOS]
| PHP version | [8.3]
| PHP_CodeSniffer version | [3.7.2]
| Standard | [Squiz]
| Install type | [composer local]

## Please confirm:

- x ] I have searched the issue list and am not opening a duplicate issue.
- x ] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [ ] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer. (since https://github.com/squizlabs/PHP_CodeSniffer/issues/2937 was closed as resolved I guess this is another case.)

**Comment by jrfnl** [comment]
Please don't open issues in this repo anymore. This repo is abandoned and https://github.com/PHPCSStandards/PHP_CodeSniffer is its successor. See: #3932

Also, please start by updating your PHPCS install to the latest release (3.10.1) and only report in the new repo if the issue still exists.

**Comment by simPod** [comment]
Ah sorry, got confused
